narwhals/_pandas_like/utils.py

```diff
@@ -160,7 +160,7 @@ def broadcast_and_extract_dataframe_comparand(index: Any, other: Any) -> Any:
     if isinstance(other, PandasLikeSeries):
         len_other = other.len()
 
-        if len_other == 1:
+        if len_other == 1 and len(index) != 1:
```

Comment:
Similarly as for pyarrow, otherwise for list/array types we end up getting the first element